modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -323,16 +323,40 @@ class QueueReader
 void rewindToStop(std::vector<Q*> &in_queues,
                   const std::size_t  this_id)
 {
-    for (auto &&qit : ade::util::indexed(in_queues))
-    {
-        auto id2 = ade::util::index(qit);
-        auto &q2 = ade::util::value(qit);
-        if (this_id == id2) continue;
+    size_t expected_stop_count = std::count_if(in_queues.begin(), in_queues.end(), [] (const Q* ptr) {
+        return ptr != nullptr;
+    });
 
-        Cmd cmd;
-        while (q2 && !cv::util::holds_alternative<Stop>(cmd))
-            q2->pop(cmd);
+    if (expected_stop_count > 0) {
+        // NB: it requires to substract own queues id from total waiting queue count
+        // because it had got stop message before rewind was called
+        expected_stop_count--;
+    }
+    GAPI_LOG_DEBUG(nullptr, "id: " << this_id << ", queues count: " << in_queues.size() <<
```

Comment:
Not sure that it's can somehow help user

Reply:
This is still a debug message, we have some already so I'd not care about this one

Reply:
Got it, then @sivanov-work  you can ignore these comments :)